components-api/src/main/scala/pl/touk/nussknacker/engine/api/process/ProcessObjectDependencies.scala

```diff
@@ -1,15 +1,39 @@
 package pl.touk.nussknacker.engine.api.process
 
 import com.typesafe.config.Config
+import net.ceedubs.ficus.Ficus.toFicusConfig
+import net.ceedubs.ficus.readers.AnyValReaders._
+import net.ceedubs.ficus.readers.OptionReader._
 import pl.touk.nussknacker.engine.api.namespaces.NamingStrategy
+import pl.touk.nussknacker.engine.api.process.ProcessObjectDependencies.ModelConfig
 
 // TODO: Rename to ModelDependencies + rename config to modelConfig
-final case class ProcessObjectDependencies private (config: Config, namingStrategy: NamingStrategy) extends Serializable
+final case class ProcessObjectDependencies private (modelConfig: ModelConfig, namingStrategy: NamingStrategy)
+    extends Serializable {
+  def config: Config = modelConfig.underlyingConfig
+}
 
 object ProcessObjectDependencies {
 
-  def withConfig(modelConfig: Config): ProcessObjectDependencies = {
-    ProcessObjectDependencies(modelConfig, NamingStrategy.fromConfig(modelConfig))
+  def apply(underlyingConfig: Config, namingStrategy: NamingStrategy): ProcessObjectDependencies = {
```

Comment:
Let's use `ModelConfig` here and in `withConfig` method. Generally, let replace  `Config` with `ModelConfig` everywhere where it is easy. Here it is.